src/librustc/error_codes.rs

```diff
@@ -1956,6 +1956,21 @@ a (non-transparent) struct containing a single float, while `Grams` is a
 transparent wrapper around a float. This can make a difference for the ABI.
 "##,
 
+E0697: r##"
+A closure has been used as `static`.
+
+Erroneous code example:
+
+```compile_fail,E0697
+fn main() {
+    static || {}; // used as `static`
+}
+```
+
+Closures cannot be used as `static`. The easist fix is to remove `static`
+keyword.
```

Comment:
Also, might be interesting to explain why it's "bad" to use a closure as a static.

Reply:
Hmm, I can't come up with an explanation of why. Any documentation or you have an explanation?

Reply:
It's more about the closures themselves: they "save" the environment. Therefore, having a static closure with a static environment doesn't really make sense since all you can capture inside it would be variables with static lifetime. Quite limiting. In this condition, better use a function directly.

Reply:
Thanks! I added your words.